src/d3a/d3a_core/live_events.py

```diff
@@ -46,6 +49,20 @@ def __init__(self, area_uuid, area_params):
     def apply(self, area):
         if area.uuid != self.area_uuid:
             return False
+        if key_in_dict_and_not_none(self.area_params, 'type'):
+            if self.area_params['type'] == "MarketMaker":
+                strategy_obj = MarketMakerStrategy
+                del self.area_params['type']
```

Comment:
This line can be deleted. type is also removed from the dict couple lines below

Reply:
Actually I have deleted from there and have to keep it here else the new strategy initialisation would complain as `type` parameter is unknown to it and ATM strategy doesn't accept arbitrary args and kwargs either.

Reply:
Why are you handling `type` differently from e.g. `energy_rate_profile_uuid ` (meaning the difference between `del` and `pop`?

Reply:
Also, if you are using pop, there could be only one place where all these parameters get deleted (exactly after `if key_in_dict_and_not_none(self.area_params, 'type'):`). There could even be a small function called, for instance, sanitize_market_maker_dict that could remove all these parameters from the dict regardless of the type value. Then you could also keep the type as a variable, and use this for the if/else blocks.